### DISCUSSION TITLE
How to unlink users from LDAP Federation

### DISCUSSION THREAD

**Discussion by mozdi** [keycloak/keycloak#24059] [Q&A]
Because our federated LDAP Server will be decomissioned soon, we need to migrate users from a federated ldap to the local user database of keycloak. I understand, that it´s not possible to fully import the users including passwords.

One possible option to migrate the federated users is to unlink them from federation and ask them to set a new password.
Unfortunately, in my instance, the option to "unlink users" is greyed out in the "Action"-drop-down in the federation and I´m not able to remove the users link.
![image](https://github.com/keycloak/keycloak/assets/39409442/82c63adb-f25e-4f4e-9095-42f08e19dd08)

Is there any option I need to set in the configuration to enable the unlink-option in the federation?

**Comment by jonkoops** [comment]
It looks like this won't be possible unless "Import users" is enabled in the LDAP settings. If you believe this is a bug, please report it so we can triage this.

**Reply by mozdi** [reply]
@jonkoops, the option should be available if "Import users" is enabled in the ldap federation?
If so, it seems to be a bug, because "import users" is enabled and the option is still greyed out.

**Reply by jonkoops** [reply]
Please do report this issue as a bug then.

**Reply by mozdi** [reply]
Thanks!
I‘ve submitted https://github.com/keycloak/keycloak/issues/24261
